aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OutboxMutationFailedEvent.java

```diff
@@ -111,8 +96,83 @@ public M getModel() {
         return model;
     }
 
+    /**
+     * Gets the name of the model.
+     * @return the model name
+     */
+    @NonNull
+    public String getModelName() {
+        return model.getClass().getSimpleName();
```

Comment:
This won't work when we are using `ModelSchema` on flutter. The new approach is to always ask `modelSchema.getName()`. I had to update this logic in https://github.com/aws-amplify/amplify-android/issues/1012. You can most likely use a similar technique to access the model name externally, and then pass it in when you are constructing the `OutboxMutationFailedEvent`.

TLDR, `getClass().getSimpleName()` isn't a thing for us anymore, we have to make them all say `modelSchema.getName()`.